src/server/controller/plugin.ts

```diff
@@ -0,0 +1,56 @@
+import { PostgresGateway } from "../gateways/index";
+import { Trade } from "../types";
+// import memcached from 'memcached';
+
+export default class Controller {
+  db: PostgresGateway;
+
+  constructor() {
+    this.db = new PostgresGateway();
+  }
+
+  public async write(body: Trade): Promise<{}> {
+    try {
+      const id = await this.db.write(body);
+      return id;
+    } catch (error) {
+      throw error;
```

Comment:
**Comment**: Errors should also be logged at this level, otherwise in cases where we unsuccessfully parse the successful result of a database query will not be properly caught.

Reply:
Added error logger to all controller methods.